src/valid/function.rs

```diff
@@ -805,6 +812,40 @@ impl super::Validator {
                 } => {
                     self.validate_atomic(pointer, fun, value, result, context)?;
                 }
+                S::WorkgroupUniformLoad { pointer, result } => {
+                    if self.valid_expression_set.insert(result.index()) {
+                        self.valid_expression_list.push(result);
+                    } else {
+                        return Err(FunctionError::ExpressionAlreadyInScope(result)
+                            .with_span_handle(result, context.expressions));
+                    }
+                    match &context.expressions[result] {
+                        &crate::Expression::WorkgroupUniformLoadResult { pointer: inner_ptr }
+                            if pointer == inner_ptr => {}
+                        _ => {
+                            return Err(FunctionError::WorkgroupUniformLoadExpressionMismatch(
+                                result,
+                            )
+                            .with_span_handle(result, context.expressions))
+                        }
+                    }
+                    match context.resolve_type(pointer, &self.valid_expression_set)? {
+                        &Ti::Pointer { base, .. }
+                            if self.types[base.index()]
+                                .flags
+                                // HACK: The set of types for SIZED|CONSTRUCTIBLE
+                                // is, as far as I can tell, exactly the types required (i.e. fixed-size data types without atomic types)
```

Comment:
Not a hack, this is correct. `CONSTRUCTIBLE` actually implies `SIZED`, so it could be omitted.